numpy/core/src/umath/ufunc_object.c

```diff
@@ -2216,26 +2244,50 @@ PyUFunc_GeneralizedFunction(PyUFuncObject *ufunc,
     /* Get all the arguments */
     retval = get_ufunc_arguments(ufunc, args, kwds,
                 op, &order, &casting, &extobj,
-                &type_tup, &subok, NULL, &axes);
+                &type_tup, &subok, NULL, &axes, &axis, &keepdims);
     if (retval < 0) {
-        goto fail;
```

Comment:
The change from `goto fail` to `return` here and below is because in `fail` I now need to decref axes (since it might be a new object created from `axis`). Fortunately, we have not taken any references here (and below), so it is safe to just return.